pkg/sm/sm.go

```diff
@@ -130,21 +133,46 @@ func New(ctx context.Context, cancel context.CancelFunc, e env.Environment, cfg
 
 	// Setup core API
 	log.C(ctx).Info("Setting up Service Manager core API...")
-
 	pgNotificator, err := postgres.NewNotificator(smStorage, cfg.Storage)
 	if err != nil {
 		return nil, fmt.Errorf("could not create notificator: %v", err)
 	}
 
+	blockedClientsManager := blocked_clients.Init(ctx, interceptableRepository)
+	callbacks := map[string]func(*events.Message) error{
+		"blocked_clients-INSERT": func(envelope *events.Message) error {
+			var blockedClient types.BlockedClient
+
+			if err := json.Unmarshal(envelope.Data, &blockedClient); err != nil {
+				log.C(ctx).Info("error unmarshaling new blocked client")
+			}
```

Comment:
maybe i dont get your logic but if there is an error you are still adding to the cache the client ... ?